lib/segment/src/segment_constructor/segment_builder.rs

```diff
@@ -191,6 +191,14 @@ impl SegmentBuilder {
                     }
                     break;
                 }
+                FieldIndex::UuidIndex(index) => {
+                    if let Some(ids) = index.get_values(internal_id) {
+                        for id in ids {
+                            ordering = ordering.wrapping_add(id as u64);
```

Comment:
It looks like i128 -> u64 conversion goes by lower bytes. In case of UUID it might be better to order by higher, more valuable bytes.

Experiments: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bfaa47e0247696885bf6d7a93f1b66aa

Reply:
Are you sure, the first bits of a Uuid aren't the ones storing the timestamp?
Given the table in [Wikipedia](https://en.wikipedia.org/wiki/Universally_unique_identifier#Binary_wire_format) and the uuid [crates source code](https://docs.rs/uuid/latest/src/uuid/lib.rs.html#627), it seems to be the case as well.
Please correct me if I'm reading those sources wrong.

Reply:
The alternative would be to change ordering value to 128 bit

Reply:
Also ordering by timestamp seems logical, if we have option between ordering by random or by timestamp, I would prefer timestamp

Reply:
Yeah I agree. I'll update it to using `get_timestamp` (and  `as_fields` as fallback).